contrib/win32/win32compat/misc.c

```diff
@@ -314,22 +363,35 @@ spawn_child(char* cmd, int in, int out, int err, DWORD flags) {
 	si.dwFlags = STARTF_USESTDHANDLES;
 
 	b = CreateProcessW(NULL, cmd_utf16, NULL, NULL, TRUE, flags, NULL, NULL, &si, &pi);
+	GOTO_CLEANUP_IF(!b, ENOEXEC);
+	CloseHandle(pi.hThread);
```

Comment:
before your code changes, we used to execute this "CloseHandle(pi.hThread);" only when register_child() fails but now we are doing in all the cases.. Did you check if it causes any problem?